docs: Link to pyo3-log in README and the guide (#1052)
PyO3/pyo3: guide/src/logging.md

# Logging

It is desirable if both the Python and Rust parts of the application end up
logging using the same configuration into the same place.

> This can trivially be accomplished by not logging at all in one of them. If

Reviewer: I think this sentence doesnt add so much - users presumably come here because they want to do something rather than not log at all :smile: 

I'm tempted to suggest this paragraph gets replaced with a new one that describes what the rest of this chapter will be, e.g.

```suggestion
This section of the guide briefly discusses how to connect the two languages' logging ecosystems together. The recommended way for Python extension modules is to configure Rust's logger to send log messages to Python using the `pyo3-log` crate. For users who want to do the opposite and send Python log messages to Rust, see the note at the end of this guide.
```

This will help readers find what section of this chapter could be relevant to them by just scanning the introduction.
 
Of course please feel free to change what's written here it you don't like this suggestion!

that's not a good enough solution, it is possible to install a logger that sends
the log messages to the counterpart.

> # The Rust -> Python direction

Reviewer: I wonder if this title would be even stronger if it said what the reader will learn in this section:

```suggestion
## Using `pyo3-log` to send Rust log messages to Python
```

Also I would prefer only have one top level `#` heading so I added an extra `#` in the suggestion.

> The [pyo3-log] allows sending the messages from the Rust side to Python's

Reviewer: ```suggestion
The [pyo3-log] crate allows sending the messages from the Rust side to Python's
```

[logging] system. This is mostly suitable for writing native extensions for
Python programs.

> The [`pyo3_log::init`][init] installs the logger in its default configuration.

Reviewer: ```suggestion
Use [`pyo3_log::init`][init] to install the logger in its default configuration.
```

> It's also possible to tweak it's configuration (mostly to tune its performance).

Reviewer: ```suggestion
It's also possible to tweak its configuration (mostly to tune its performance).
```

```rust
use log::info;
use pyo3::prelude::*;
use pyo3::wrap_pyfunction;

#[pyfunction]
fn log_something() {
    // This will use the logger installed in `my_module` to send the `info`
    // message to the Python logging facilities.
    info!("Something!");
}

#[pymodule]
fn my_module(_py: Python<'_>, m: &PyModule) -> PyResult<()> {
    // A good place to install the Rust -> Python logger.

>     pyo3_log::init();

Reviewer 2: Nice interface :+1:

    m.add_wrapped(wrap_pyfunction!(log_something))?;
    Ok(())
}
```

Then it is up to the Python side to actually output the messages somewhere.

```python
import logging
import my_module

FORMAT = '%(levelname)s %(name)s %(asctime)-15s %(filename)s:%(lineno)d %(message)s'
logging.basicConfig(format=FORMAT)
logging.getLogger().setLevel(logging.INFO)
my_module.log_something()
```

It is important to initialize the Python loggers first, before calling any Rust
functions that may log. This limitation can be worked around if it is not

> possible to satisfy, read the documentation about [caching]

Reviewer: ```suggestion
possible to satisfy, read the documentation about [caching].
```

> [logging]: https://docs.python.org/3/library/logging.html

Reviewer: How about maybe adding a very short section about the `Python -> Rust` direction - that nobody has built a solution yet and if would be awesome if anyone wants to contribute a PR / crate for that (with design advice available from the PyO3 community)?
